### DISCUSSION TITLE
Phlex::Icons gem

### DISCUSSION THREAD

**Discussion by AliOsm** [ruby-ui/ruby_ui#101] [Show and tell]
Hello everyone,

I've been migrating one of my projects to Phlex/PhlexUI over the past two weeks and realized I frequently use SVG icons from various sources like Heroicons, Bootstrap icons, and Flag icons. Instead of manually integrating these icons into a `Shared::Icon` component, I thought, why not build a gem that offers easy-to-use Phlex components for these popular icon packs?

Initially, I came across the [phlex-heroicons](https://github.com/nejdetkadir/phlex-heroicons) gem, which provides Phlex components for Heroicons. It was exactly what I needed but only for Heroicons.

So, I decided to create the [phlex-icons](https://github.com/AliOsm/phlex-icons) gem. This gem generates Phlex components for icons from multiple sources automatically using the [phlexing](https://github.com/marcoroth/phlexing) gem. Currently, `phlex-icons` includes over 4,000 icons from four different sources:

1. Heroicons (300+)
2. Lucide icons (1,500+)
3. Bootstrap icons (2,000+)
4. Flag icons (250+)

The goal of `phlex-icons` is to provide developers with a centralized, easy-to-integrate solution for incorporating a wide range of icons into their Phlex/PhlexUI projects. By automating the generation of icon components, this gem can save a significant amount of time and effort.

As a next step, I'm planning to set up a GitHub Actions workflow to automatically update the icon packs weekly, ensuring that users always have access to the latest icons.

I'd love to hear your thoughts on this project! Are there other icon packs you’d like to see included? Any features or improvements you think would be valuable? Please feel free to try out the gem, share your feedback, and contribute to its development.

Thank you!

**Comment by cirdes** [comment]
@AliOsm, congratulations on the gem! It is really useful for working with icons in the Rails universe. For PhlexUI/RbUI, it's better to work directly with the SVGs to avoid additional dependencies and keep things simple.

For the website, we're using the `lucide-rails` gem, which I'm helping to maintain.

https://github.com/PhlexUI/PhlexUI-web/blob/18546222ec35b62ae53211d43c16344ad520b894/Gemfile#L25

**Reply by AliOsm** [reply]
I noticed the use of lucide icons gem in the webapp, that is why I added them to phlex-icons, as I didn't know about them before, they are really beautiful.

In general, do you think the phlex-icons idea is a good one? I mean, having 4,000+ classes 😁

**Reply by cirdes** [reply]
@AliOsm , PhlexUI is inspired by shadcn and shadcn uses Lucide! 😄 

I do think it's a great idea and more important, Joel seems to agree with you!  

https://github.com/orgs/phlex-ruby/discussions/756#discussioncomment-10507043

**Reply by AliOsm** [reply]
Oh I didn't realize that Joel replied :3 
Let's see what will happen :) Thanks for the support!

**Reply by AliOsm** [reply]
I just noticed the existence of [phlex-lucide](https://github.com/akodkod/phlex-lucide), if PhlexUI-Web is interested :3

**Comment by AliOsm** [comment]
If you don't like the all in one solution for any reason, `phlex-icons` is now available as a separate gem per icon pack, you just need to append `-{icon pack name}` to the gem name and you will get it (e.g. `phlex-icons-lucide`) 🎉

![‏لقطة الشاشة ٢٠٢٤-٠٩-٠٢ في ١٢ ٤٣ ٤٣ ص](https://github.com/user-attachments/assets/0f78e3b3-c522-48e5-b07c-38e5399c9dfc)

**Comment by konung** [comment]
This is amazing. I was just looking at various packs - trying to figure out which one is the "most" complete - because I didn't want to spend time to add each one individually  - and found this issue by accident. @AliOsm  - great job!

**Reply by AliOsm** [reply]
Most welcome, more packs are in the way 😁

**Reply by konung** [reply]
More? 🥇  I didn't know about 3 packs you had here, what else is even out there ? (Bewildered)

**Reply by AliOsm** [reply]
I'm publishing a new version with Hugeicons in seconds. Also, font awesome, iconoir, feather, boxicons, ionicons, and others are in the plan.
